SLAM/iterative_closest_point/iterative_closest_point.py

```diff
@@ -124,6 +118,31 @@ def svd_motion_estimation(previous_points, current_points):
     return R, t
 
 
+def plot_points(previous_points, current_points, figure):
+    # for stopping simulation with the esc key.
+    plt.gcf().canvas.mpl_connect(
+        'key_release_event',
+        lambda event: [exit(0) if event.key == 'escape' else None])
+    if previous_points.shape[0] == 3:
+        # axes = figure.axes[0]
+        # axes.clear()
+        figure = plt.figure()
```

Comment:
I'm not sure why are you creating figure every iteration?
This works without figure creation in my environment.
```suggestion
                plt.clf()
```

Reply:
For 3d case in Spyder I can see only the first plot with initial point position. And it is not updating.
(Need to mention that for 2d case every single frame of optimization is displayed as a separate image, and not an updated previous)
May be this is the case for Spyder. Does it work for you without creating figure for 3d?

Reply:
Yeah. In my environment (just run the script from a terminal), it works. I think creating a lot of figures is annoying. Could you try to run the script from terminal?